### PR TITLE
updates the Makefile to ensure consistent installations of black and click.

### PR DESCRIPTION
**Pull request by Akshat2Jain**
## Description

This PR updates the Makefile to ensure consistent installations of black and click. Currently, the Makefile installs the latest versions, while the CI process installs specific versions mentioned in the versions.cfg file. This PR aligns the installations in the Makefile with the versions specified in the versions.cfg file.

## Changes Made

- Modified the Makefile to install black and click with the versions mentioned in versions.cfg.
OR
- Upgraded the installations of black and click to the latest versions mentioned in versions.cfg.

## Reason for the Changes

- Ensures consistency between local development and the CI environment.
- Aligns installations with the versions specified in versions.cfg.
OR
- Benefits from the latest features and bug fixes available in the upgraded versions.

This is pr is realated to #1655

### CODE REVIEW

**Comment by Akshat2Jain** [comment]
@wesleybl Plz review

File: news/1668.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+updates the Makefile to ensure consistent installations of black and click. @Akshat2Jain
```
Comment (wesleybl) [inline news/1668.bugfix:1]: Use the issue number in the file name. Use the internal extension.

File: Makefile [hunk 2]
```diff
@@ -44,7 +44,7 @@ update: ## Update Make and Buildout
 bin/buildout: bin/pip
 	bin/pip install --upgrade pip
 	bin/pip install -r requirements-5.2.txt
-	bin/pip install black || true
+	bin/pip install --upgrade click==$(awk '/^click =/{print $NF}' versions.cfg) black==$(awk '/^black =/{print $NF}' versions.cfg) || true
```
Comment (wesleybl) [inline Makefile:47]: Looks like black wasn't installed:

```shell
$ make black
if [ -f "bin/black" ]; then bin/black src/ ; fi
```

**Review by wesleybl — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (wesleybl) [reply Makefile:47]: Take a look at:

https://stackoverflow.com/questions/30445218/why-does-awk-not-work-correctly-in-a-makefile

File: news/1668.bugfix (hunk 1 above)
Comment (Akshat2Jain) [reply news/1668.bugfix:1]: Sorry but I'm not familiar with that term . what exactly you meant by 'internal extension.'

Comment (wesleybl) [reply news/1668.bugfix:1]: In the same way that we have the `.bugfix`, we also have the `.internal`. The options are here:

https://github.com/plone/plone.restapi/blob/master/pyproject.toml

We use `.internal`, when the change is interest only to the package developers.

Comment (stevepiercy) [reply news/1668.bugfix:1]: @Akshat2Jain see also https://6.docs.plone.org/contributing/index.html#change-log-entry

Comment (Akshat2Jain) [reply news/1668.bugfix:1]: there is no issue , so should I name the file as `1668.internal`

File: Makefile (hunk 2 above)
Comment (Akshat2Jain) [reply Makefile:47]: > Looks like black wasn't installed:
> 
> ```shell
> $ make black
> if [ -f "bin/black" ]; then bin/black src/ ; fi
> ```

but it shows black version `23.3.0`, so it means black is installed

Comment (wesleybl) [reply Makefile:47]: It must have been installed before you update the code in the Makefile. You can run a `make clean` to remove the previous installation. Then if you run `make` again you will see that `black` is not installed. The error occurs:]

```
bin/pip install --upgrade click== black== || true
ERROR: Could not find a version that satisfies the requirement click== (from versions: 0.1, 0.2, 0.3, 0.4, 0.5, 0.5.1, 0.6, 0.7, 1.0, 1.1, 2.0, 2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 3.0, 3.1, 3.2, 3.3, 4.0, 4.1, 5.0, 5.1, 6.0, 6.1, 6.2, 6.3, 6.4, 6.5, 6.6, 6.7.dev0, 6.7, 7.0, 7.1, 7.1.1, 7.1.2, 8.0.0a1, 8.0.0rc1, 8.0.0, 8.0.1, 8.0.2, 8.0.3, 8.0.4, 8.1.0, 8.1.1, 8.1.2, 8.1.3, 8.1.4)
ERROR: No matching distribution found for click==
```

**Comment by Akshat2Jain** [comment]
Hey @wesleybl, I have made the changes, now it installed the correct version from the version.cfg
```
$ ./bin/black --version
black, 22.3.0 (compiled: yes)
```

**Comment by Akshat2Jain** [comment]
Hey @stevepiercy, I have changed the file name to `1668.internal `

File: news/1668.internal [hunk 3]
```diff
@@ -0,0 +1 @@
+updates the Makefile to ensure consistent installations of black and click. @Akshat2Jain
```
Comment (stevepiercy) [inline news/1668.internal:1]:
```suggestion
Updated the Makefile to ensure consistent installations of black and click. @Akshat2Jain
```

**Comment by stevepiercy** [comment]
There's a failing test, and I have no idea why it fails: https://github.com/plone/plone.restapi/actions/runs/5529801801/jobs/10088359044?pr=1668#step:10:205

**Review by stevepiercy — APPROVED** [review]

**Comment by stevepiercy** [comment]
And now the failing test passes. `¯\_(ツ)_/¯`

**Review by wesleybl — APPROVED** [review]
LGTM

**Comment by tisto** [comment]
LGTM. Though, I am afraid we currently do not use versions.cfg as far as I can see. I'd suggest including it everywhere and cleaning it up. Or we just use base.cfg to fetch the black versions (seems like the easier way to go for now).

**Comment by wesleybl** [comment]
@tisto you are sure. `versions.cfg` is not used by buildout. So I'm +1 for revomering `versions.cfg`.

Looking closer, the buildout doesn't install `black`. Only the Makefile and the CI. So I think it would be better to create a constraints.txt, and use it in these places. So the install would just be something like:

```
pip install -c constraints.txt black
```

This simplifies the installation avoiding complications like:

https://github.com/plone/plone.restapi/blob/9097cfcb55a7b2a045a5c5e66c46cd04300798c2/.github/workflows/black.yml#L25

**Review by wesleybl — CHANGES_REQUESTED** [review]
See: https://github.com/plone/plone.restapi/pull/1668#issuecomment-1633424332

Maybe it's better to create another PR.

**Comment by wesleybl** [comment]
Closed in favor of #1671.

**Pull request closed without merging**